js/memory.js

```diff
@@ -2,8 +2,82 @@ class MemoryGame {
   constructor(cards){
     this.cards = cards;
     // add the rest of the class properties here
+    this.pickedCards = [];
+    this.pairClicked = [];
+    this.pairsGuessed = [];
   }
-  shuffleCards() {}
-  checkIfPair(card1, card2) {}
-  isFinished() {}
-}
\ No newline at end of file
+  shuffleCards() { // found this on Stack overflow.
+    function shuffle() {
+      // creating the current index based on this.cards array length.
+      var currentIndex = this.cards.length, 
+      temporaryValue, randomIndex;
+    
+      // While there remain elements to shuffle...
+      while (0 !== currentIndex) {
+    
+        // Pick a remaining element...
+        randomIndex = Math.floor(Math.random() * currentIndex);
+        currentIndex -= 1;
+    
+        // And swap it with the current element.
+        temporaryValue = array[currentIndex];
```

Comment:
It is true that you are accessing an index in an array.
However, what array are you accessing? Could it perhaps be this.cards?